var/spack/repos/builtin/packages/py-ipython/package.py

```diff
@@ -35,43 +45,55 @@ class PyIpython(PythonPackage):
+    depends_on("py-pickleshare", when="@4:", type=("build", "run"))
+    depends_on("py-prompt-toolkit@3.0.2:3.0", when="@8.5:", type=("build", "run"))
     depends_on("py-prompt-toolkit@2.0.0:2,3.0.2:3.0", when="@7.26:", type=("build", "run"))
     depends_on("py-prompt-toolkit@3.0.2:3.0", when="@7.18:7.25", type=("build", "run"))
     depends_on("py-prompt-toolkit@2.0.0:2.0", when="@7.5.0", type=("build", "run"))
     depends_on("py-prompt-toolkit@2.0.0:2", when="@7.0.0:7.5.0", type=("build", "run"))
-    depends_on("py-prompt-toolkit@1.0.4:1", when="@:7.0.0", type=("build", "run"))
-    depends_on("py-pygments", type=("build", "run"))
-    depends_on("py-backcall", when="@7.3.0:", type=("build", "run"))
+    depends_on("py-prompt-toolkit@1.0.4:1", when="@5:7.0.0", type=("build", "run"))
+    depends_on("py-prompt-toolkit@1.0.3:1", when="@5:7.0.0", type=("build", "run"))
+    depends_on("py-pygments@2.4:", when="@8.1:", type=("build", "run"))
+    depends_on("py-pygments", when="@5:", type=("build", "run"))
     depends_on("py-stack-data", when="@8:", type=("build", "run"))
-    depends_on("py-matplotlib-inline", when="@7.23:", type=("build", "run"))
-    depends_on("py-pexpect@4.4:", when="@7.18: platform=linux", type=("build", "run"))
-    depends_on("py-pexpect@4.4:", when="@7.18: platform=darwin", type=("build", "run"))
-    depends_on("py-pexpect@4.4:", when="@7.18: platform=cray", type=("build", "run"))
-    depends_on("py-pexpect", type=("build", "run"))
-    depends_on("py-appnope", when="platform=darwin", type=("build", "run"))
-    depends_on("py-colorama", when="platform=windows", type=("build", "run"))
+    depends_on("py-traitlets@5:", when="@8:", type=("build", "run"))
+    depends_on("py-traitlets@4.2:", when="@5:", type=("build", "run"))
+    depends_on("py-traitlets", when="@4:", type=("build", "run"))
 
-    depends_on("py-backports-shutil-get-terminal-size", when="^python@:3.2", type=("build", "run"))
-    depends_on("py-pathlib2", when="^python@:3.3", type=("build", "run"))
-    depends_on("py-simplegeneric@0.8:", when="@:7.0.0", type=("build", "run"))
+    depends_on(
+        "py-backports-shutil-get-terminal-size", when="@4.2: ^python@:3.2", type=("build", "run")
+    )
+    depends_on("py-black", when="@8.0", type=("build", "run"))
+    depends_on("py-pathlib2", when="@5: ^python@:3.3", type=("build", "run"))
+    depends_on("py-simplegeneric@0.8:", when="@4:7.0.0", type=("build", "run"))
+    depends_on(
+        "py-win-unicode-console@0.5:",
+        when="@5:7.9 ^python@:3.5 platform=windows",
+        type=("build", "run"),
+    )
 
     @property
-    def import_modules(self):
-        modules = super(__class__, self).import_modules
+    def skip_modules(self):
 
         if self.spec.satisfies("@8:"):
-            return modules
+            return []
 
         # 'IPython.kernel' is deprecated and fails to import, leave out of
         # 'import_modules' to ensure that import tests pass.
```

Comment:
Trying to understand this logic. In IPython 7 and older, IPython.kernel was deprecated, so we skip that import. But in IPython 8 it is no longer deprecated and imports fine? Or it no longer exists so there's no reason to skip it? Maybe we can just always skip it if it's the latter.

Reply:
Let's just always return `IPython.kernel`, not returning anything when it doesn't exist seems like an unnecessary step. Alternatively, we can only skip it after it becomes deprecated.

Reply:
Done.